docs/contributing.rst

```diff
@@ -62,26 +62,176 @@ The following Python implementations are supported:
+      instrumentation for an example.
+
+    - Gathering relevant information: ``ddtrace`` provides a helper for pulling
+      out this information and adding it to a span.  See the `cassandra
+      exception handling
+      <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/cassandra/session.py#L117-L122>`_
+      instrumentation for an example.
+
+
+Distributed Tracing
+~~~~~~~~~~~~~~~~~~~
+
+Some integrations can propagate a trace across application boundaries to other
+applications where the trace is continued. Datadog and ``ddtrace`` provide
+support for continuing a trace in another application.
+
+    - Propagating the trace example: `requests <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/requests/connection.py#L85-L88>`_
+    - Receiving a propagated trace example: `Django <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/django/middleware.py#L116-L121>`_
+
+
+Web frameworks
+++++++++++++++
+
+
+A web framework integration must do the following if possible:
+
+    - Install the WSGI or ASGI trace middlewares already provided by ``ddtrace``.
+    - Trace the duration of the request.
+    - Assign a resource name for a route.
+    - Use ``trace_utils.set_http_meta`` to set the standard http tags.
+    - Have an :ref:`internal service` name.
+    - Support distributed tracing (configurable).
+    - Provide insight to middlewares and views.
+    - Use the `SpanTypes.WEB` span type.
+
+Some example web framework integrations::
+    - `flask <https://github.com/DataDog/dd-trace-py/tree/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/flask>`_
+    - `django <https://github.com/DataDog/dd-trace-py/tree/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/django>`_
+
+
+Database libraries
+++++++++++++++++++
+
+``ddtrace`` already provides base instrumentation for the Python database API
+(PEP 249) which most database client libraries implement in the
+`ddtrace.contrib.dbapi <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/dbapi/__init__.py>`_
+module.
+
+Check out some of our existing database integrations for how to use the `dbapi`:
+
+    - `psycopg <https://github.com/DataDog/dd-trace-py/tree/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/psycopg>`_
+    - `mysql <https://github.com/DataDog/dd-trace-py/tree/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/mysql>`_
+
+
+Testing
++++++++
+
+The tests must be defined in its own module in ``tests/contrib/<integration>/``.
+
+Testing is the most important part of the integration. We have to be certain
+that the integration:
+
+    1) works: submits meaningful information to Datadog
+
+    2) is invisible: does not impact the library or application by disturbing state,
+       performance or raising exceptions
+
```

Comment:
Maybe add a line suggesting to look at an integration that's similar to the one you're working on for reference on test creation?